torch/testing/_internal/common_device_type.py

```diff
@@ -311,11 +315,11 @@ def instantiated_test(self, name=name, test=test_fn, dtype=dtype, op=op):
                 instantiate_test_helper(cls, name, test=test, dtype=dtype, op=None)
 
 
-class CPUTestBase(DeviceTypeTestBase):
+class CPUTestBase(Generic[DeviceTypeTestBase]):
     device_type = 'cpu'
 
 
-class CUDATestBase(DeviceTypeTestBase):
+class CUDATestBase(Generic[DeviceTypeTestBase]):
```

Comment:
similarly for example
```
Type[CUDATestBase]" has no attribute "primary_device"
``` 
means you need to pre-define `primary_device: ClassVar[str]` under this `CUDATestBase` class so that mypy knows there's a class attribute `primary_device` of type String